### ISSUE TITLE
error in csv file opening 

### ISSUE THREAD

**Issue by ravijoe** [lyst/lightfm#292]
so this here is my code .

import csv
import numpy as np
import matplotlib.pyplot as plt
from sklearn.svm import SVR
dates=[]
prices=[]
def get_data(filename):
    with open(filename,'r') as csvfile:
        csvFileReader=csv.reader(csvfile)
        next(csvFileReader)
        for row in csvFileReader:
            dates.append(int(row[0].split('-')[0]))
            prices.append(float(row[1]))
    return

def predict(dates,prices,x):
    dates=np.reshape(dates,(len(dates),1))
    svr_lin=SVR(kernel='linear',C=1e3)
    svr_poly=SVR(kernel='poly',C=1e3,degree=2)
    svr_rbf=SVR(kernel='rbf',C=1e3,gamma=0.1)
    svr_lin.fit(dates,prices)
    svr_poly.fit(dates, prices)
    svr_rbf.fit(dates, prices)
    
    plt.scatter(dates,prices,color='black',label='Data')
    plt.plot(dates,svr_rbf.predict(dates),color='red',label='RBF model')
    plt.plot(dates, svr_lin.predict(dates), color='green', label='Linear model')
    plt.plot(dates, svr_poly.predict(dates), color='blue', label='Polynomial model')
    plt.xlabel('Date')
    plt.ylabel('Price')
    plt.title('Support Vector Regression')
    plt.legend()
    plt.show()
    return svr_rbf.predict(x)[0],svr_lin.predict(x)[0],svr_poly.predict(x)[0]

get_data('aapl.csv')
predicted_price=predict(dates,prices,29)
print(predicted_price)

and i am getting following errors

![capture1](https://user-images.githubusercontent.com/29393328/39629462-ea5c8b48-4fc9-11e8-964b-10d55048b80e.JPG)

and here is the 'aapl.csv ' file
[aapl.zip](https://github.com/lyst/lightfm/files/1974498/aapl.zip)

**Comment by maciejkula** [comment]
You're code is buggy. It looks like you may have more luck splitting your lines on tabs rather than hyphens.

**Comment by ravijoe** [comment]
dude . why do you always have to close the thread without even properly answering it ?

**Comment by maciejkula** [comment]
Hey @ravijoe, I'll be happy to help you with LightFM issues. So far, it looks like your problems are more of a general programming nature: how to install packages, how to read CSV files. For questions like these, sites like StackOverflow may be more useful.

**Comment by ravijoe** [comment]
i asked my question here because stackoverflow also couldn't answer the question . could you please run this code by yourself ?

**Comment by maciejkula** [comment]
It looks like you have (at least) two problems with your code.

- The CSV file you're reading is tab-, not comma-separated. The Python `csv` package has options to deal with that.
- The dates are `\`-separated, not `-`-separated.

This is quite clear from the error message you posted: the line from the file that is printed clearly shows both things.

**Comment by ravijoe** [comment]
ok. so the problem is only in the csv file right ? what i meant by earlier comment was that you fix the errors and try to run the code so that we can be sure there are no other errors . pus this is a fun stock market predictor tool.

**Comment by maciejkula** [comment]
@ravijoe, it's OK to ask for help when you get stuck --- but asking people to do your work for you is not.

I would recommend finding a contractor to help you with your project, and paying them.

**Comment by ravijoe** [comment]
you got it all wrong man . suggested edits in a code along with an explanation does not amount to asking someone to do my project . nevermind the question . 
have a good day .
